src/plugins/intel_gpu/src/graph/include/program_node.h

```diff
@@ -494,6 +497,35 @@ struct typed_program_node_base : public program_node {
         return std::static_pointer_cast<const PType>(program_node::get_primitive());
     }
 
+    size_t hash() const override {
+        if (!seed) {
+            // hash for primitive
+            seed = get_primitive()->hash();
+
+            // hash for type
+            primitive_id type_str = get_primitive()->type_string();
```

Comment:
Why this is not a part of primitive hash?

Reply:
moved it.